### ISSUE TITLE
Changing options of a select command

### ISSUE THREAD

**Issue by Renkineko** [swisnl/jQuery-contextMenu#284]
Hi,

I would like to know if it's possible to change, after the initialization of the menu, the options of a select command ? Here is the behaviour I would like to implement : 

1) Initialize menu with "fixed" items (clickable elements) + an empty select (well, not empty, with a default value like "none")
2) On menu event "show", use an Ajax Request to know the users connected to the application
3) On the success of the Ajax Request, the select is populated

The select is the same for all elements so it's in the good logic of "no multiple instance of the plugin", but I can't find any option to do it.

Context : The application manage tickets of support, and we can want to transfer some tickets to another connected user (so it can change in the lifetime of the page, I need to load them on the go). But all others items are fixed and I don't want to delay their appearance, because the transfer is less used than others fixed commands.

If I can do this with submenu instead of a select, it's even better but I think the select method is the easier ?

If impossible I'll use build design or a fixed element opening a dialog, but it's just to know if it's possible and I missed something in the documentation/on google.

**Comment by bbrala** [comment]
If you look at the example with inputs, if you use that you can use the show event to do things to your menu i think.

http://swisnl.github.io/jQuery-contextMenu/demo/input.html

     events: {
            show: function(opt) {
                // this is the trigger element
                var $this = this;
                // import states from data store 
                $.contextMenu.setInputValues(opt, $this.data());
                // this basically fills the input commands from an object
                // like {name: "foo", yesno: true, radio: "3", …}
            }, 

Check out the part in the docs about the opt argument, there is a property called inputs which might help you.

http://swisnl.github.io/jQuery-contextMenu/docs.html#opt

I'd probably say do an async ajax call there and while that is loading just make the select just contain 'loading...' or something.

**Comment by Renkineko** [comment]
I already tried to change the opt values. Here is a part of the code used for the test:

```javascript
$.contextMenu({
			"events": {
				show: function(opt) {
					opt.inputs.transferSelect.options = {
						1: 'test',
						2: 'thing'
					}
				}
			},
			"items": {
				"transferSelect": {
					name: "transfer",
					type: "select",
					options: {
						0: "loading"
					}
				}
			},
			"selector": '.ticket'
		});
```

but the input is always with the same "loading" information. I could manipulate directly the dom element containing the select but IMHO it's a very dirty hack :)

**Comment by bbrala** [comment]
Yeah the show even is after building the menu. To be honest we could make an event that fires before build or something so you can adjust options, but i read you dont want the user to wait for your call to ppulate the select. Even if there was an event beofre build, it would solve your problem since you would need the data for the select before you build and the user needs to wait then. 

I feel the thing you want to do is the reason you can access the input elements on show. So i don't really agree on the whole 'dirty hack' statement.

Just use this to access the select.

    opt.inputs.transferSelect.$input

**Comment by richpri** [comment]
I have believed for a long time that at least the opt object [and 
possibly others] should be documented
thoroughly. That is that each item and method in the object should have 
its own paragraph explaining
its purpose and its possible usage. Instead we get sometimes cryptic and 
widely scattered hints like the
one below and the ones in Issue 153 
<https://github.com/swisnl/jQuery-contextMenu/issues/153>. I would be 
willing to assist in this effort if someone with more chops
was willing to collaborate or at least serve as a resource for questions.

**Comment by bbrala** [comment]
I wouldn't mind that at all, i've been thinking about restructuring the documentation and making it a little easier to maintain. Maybe use .MD files as the base and use those to generate documentation in a more modern layout. Especially the docs.html is quite hard to find structure in.

A colleage of mine suggested maybe [couscous](http://couscous.io/), which uses the structure of MD files to generate HTML docs.

I feel this, combined with fleshing out the documentation should help a lot.

Bottom line, i would love to have help by you @richpri .

**Comment by stla** [comment]
Hello, 

 You can use [build](https://swisnl.github.io/jQuery-contextMenu/docs.html#build).
